### PR TITLE
Add get_response_local_file endpoint to API and not use AWS S3

### PR DESCRIPTION
**Pull request by ppmarkus**
I like the new feature to save as CSV but this only works with an Amazon S3 account.
Considering the file already gets saved to the server it'd be convenient if that file could be retrieved as a FileResponse directly without storing the intermediate result on the cloud. We for one are not interested in storing the created CSV files.

This feature adds a config `ONLY_STORE_CSV_FILES_LOCALLY` to set whether the user is fine only saving to S3.

Basically I would like it to work even if no S3_AWS credentials set up. This config is a quick means to check whether that can be bypassed.

Then the result can be retrieved using the a new api call `get_response_local_file` which just creates a FileResponse from the local file.

### CODE REVIEW

File: dataherald/api/fastapi.py [hunk 1]
```diff
@@ -408,6 +408,34 @@ def get_response_file(
             file_location,
             media_type="text/csv",
         )
+        
+    @override
+    def get_response_local_file(
```
Comment (jcjc712) [inline dataherald/api/fastapi.py:413]: Hi, I like your approach but what do you think about adding this logic in the current `get_response_file` method instead of creating a new one?, it could have the condition to download the file from the local path or s3 as you defined the upload in the `create_sql_query_status.py` file.
